samples/cpp/benchmark_app/main.cpp

```diff
@@ -437,6 +437,11 @@ int main(int argc, char* argv[]) {
             core.set_property(ov::cache_dir(FLAGS_cache_dir));
         }
 
+        // If set batch size, disable the auto batching
+        if (FLAGS_b > 0) {
+            core.set_property(ov::hint::allow_auto_batching(false));
```

Comment:
@ilya-lavrenov Because ov::hint::allow_auto_batching is core property and cannot be set with device name at that time. 

It is ok for me to use `core.set_property("BATCH", ov::hint::allow_auto_batching(false))` after we enable set_property with device name in master branch.

Reply:
This property is used to disable auto_batch if application has set batch size, this action is handled in core level.
 It is ok to move it into application level by applying ov::hint::allow_auto_batching() to compile_model() calling.

Reply:
> Because ov::hint::allow_auto_batching is core property

> this action is handled in core level

It's actually auto-batch property and can be passed to `core.compile_model(model, device, ov::allow_auto_batch(false))`.
Originally in 2022.1 `ov::allow_auto_batch` was not a core-level property (see [code](https://github.com/openvinotoolkit/openvino/blob/releases/2022/1/src/inference/src/ie_core.cpp))

>  after we enable set_property with device name in master branch.

It's already enabled. What is the current blocker?

I suggest two approaches how to disable auto-batch:
- `core.compile_model(model, device, ov::allow_auto_batch(false))`, which is already supported by `Core`. This is recommended approach, since it works even auto batch plugin is removed / not compiled.
- `core.set_property("BATCH", ov::allow_auto_batch(false))`. This can also be set to disable auto-batching, but currently `Core` ignores this value. Here we need to remove core-level property and update core logic to respect `BATCH` property.